File: backend/src/main/java/com/NBE3_4_2_Team4/global/exceptions/ServiceException.java
package com.NBE3_4_2_Team4.global.exceptions;

import com.NBE3_4_2_Team4.global.rsData.RsData;
import com.NBE3_4_2_Team4.standard.base.Empty;

public class ServiceException extends RuntimeException {
    private final String resultCode;
    private final String msg;

    public ServiceException(String resultCode, String msg) {
        super(resultCode + " : " + msg);
        this.resultCode = resultCode;
        this.msg = msg;
    }

    public RsData<Empty> getRsData() {
        return new RsData<>(resultCode, msg);
    }
}

File: backend/src/main/java/com/NBE3_4_2_Team4/global/exceptions/ServiceException.kt
package com.NBE3_4_2_Team4.global.exceptions

import com.NBE3_4_2_Team4.global.rsData.RsData
import com.NBE3_4_2_Team4.standard.base.Empty

class ServiceException(
    private val resultCode: String,
    private val msg: String
) : RuntimeException(
    "$resultCode : $msg"
) {
    val rsData: RsData<Empty>
        get() = RsData(resultCode, msg)
}
